File: Android/app/src/main/java/com/sans347/app/ui/components/AppChrome.kt
package com.sans347.app.ui.components

import androidx.compose.animation.animateContentSize
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.widthIn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.BarChart
import androidx.compose.material.icons.filled.Description
import androidx.compose.material.icons.filled.ExpandLess
import androidx.compose.material.icons.filled.ExpandMore
import androidx.compose.material.icons.automirrored.filled.ListAlt
import androidx.compose.material.icons.filled.Home
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.sans347.app.ui.theme.SansColors

@Composable
fun CollapsibleHeader(
    showHeader: Boolean,
    onToggleHeader: () -> Unit,
    onTables: () -> Unit,
    onHomeTitle: () -> Unit,
    onGraphs: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(SansColors.DarkBackground)
            .statusBarsPadding()
            .animateContentSize(),
    ) {
        if (showHeader) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 8.dp, vertical = 8.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Row(
                    modifier = Modifier
                        .clickable(onClick = onTables)
                        .padding(horizontal = 8.dp, vertical = 6.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Icon(Icons.Default.Description, null, tint = SansColors.Gray300, modifier = Modifier.size(16.dp))
                    Spacer(Modifier.width(4.dp))
                    Text("Tables", color = SansColors.Gray300, fontSize = 14.sp)
                }
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    modifier = Modifier
                        .weight(1f)
                        .clickable(onClick = onHomeTitle)
                        .padding(horizontal = 8.dp),
                ) {
                    Text("SANS 347", color = SansColors.White, fontWeight = FontWeight.Bold, fontSize = 18.sp)
                    Text("2024 3rd Edition", color = SansColors.PrimaryCyan, fontSize = 12.sp, fontWeight = FontWeight.Medium)
                }
                Row(
                    modifier = Modifier
                        .clickable(onClick = onGraphs)
                        .padding(horizontal = 8.dp, vertical = 6.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text("Graphs", color = SansColors.Gray300, fontSize = 14.sp)
                    Spacer(Modifier.width(4.dp))
                    Icon(Icons.Default.BarChart, null, tint = SansColors.Gray300, modifier = Modifier.size(16.dp))
                }
            }
        }
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .clickable(onClick = onToggleHeader)
                .padding(vertical = 2.dp),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = if (showHeader) Icons.Default.ExpandLess else Icons.Default.ExpandMore,
                contentDescription = null,
                tint = SansColors.Gray500,
                modifier = Modifier.size(14.dp),
            )
        }
    }
}

@Composable
fun BottomNavBar(
    currentPage: Int,
    onSelect: (Int) -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(modifier = modifier.fillMaxWidth().background(SansColors.DarkBackground)) {
        HorizontalDivider(color = SansColors.NavBorder, thickness = 1.dp)
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 6.dp, bottom = 6.dp),
            horizontalArrangement = Arrangement.SpaceAround,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            NavItem(1, currentPage, "Home", Icons.Default.Home, onSelect)
            NavItem(3, currentPage, "Results", Icons.AutoMirrored.Filled.ListAlt, onSelect)
            NavItem(2, currentPage, "Graphs", Icons.Default.BarChart, onSelect)
            NavItem(0, currentPage, "Tables", Icons.Default.Description, onSelect)
        }
    }
}

@Composable
private fun NavItem(
    page: Int,
    currentPage: Int,
    label: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    onSelect: (Int) -> Unit,
) {
    val active = currentPage == page
    val color = if (active) SansColors.PrimaryCyan else SansColors.Gray500
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = Modifier
            .widthIn(min = 60.dp)
            .clickable { onSelect(page) }
            .padding(horizontal = 8.dp, vertical = 4.dp),
    ) {
        Icon(icon, null, tint = color, modifier = Modifier.size(20.dp))
        Text(label, color = color, fontSize = 10.sp, fontWeight = FontWeight.Medium)
    }
}

File: iOS/SANS347/Views/Chrome/AppChrome.swift
import SwiftUI

/// Dark collapsible top header, mirroring `ui/components/AppChrome.kt`
/// `CollapsibleHeader`. Tables (left) / SANS 347 title (center -> Home) /
/// Graphs (right), with a chevron row that toggles the header.
struct CollapsibleHeader: View {
    let showHeader: Bool
    let onToggleHeader: () -> Void
    let onTables: () -> Void
    let onHomeTitle: () -> Void
    let onGraphs: () -> Void

    var body: some View {
        VStack(spacing: 0) {
            if showHeader {
                HStack {
                    Button(action: onTables) {
                        HStack(spacing: 4) {
                            Image(systemName: "doc.text")
                                .font(.system(size: 16))
                            Text("Tables").font(.system(size: 14))
                        }
                        .foregroundColor(SansColors.gray300)
                        .padding(.horizontal, 8).padding(.vertical, 6)
                    }
                    .buttonStyle(.plain)

                    Button(action: onHomeTitle) {
                        VStack(spacing: 0) {
                            Text("SANS 347")
                                .font(.system(size: 18, weight: .bold))
                                .foregroundColor(.white)
                            Text("2024 3rd Edition")
                                .font(.system(size: 12, weight: .medium))
                                .foregroundColor(SansColors.primaryCyan)
                        }
                        .frame(maxWidth: .infinity)
                    }
                    .buttonStyle(.plain)

                    Button(action: onGraphs) {
                        HStack(spacing: 4) {
                            Text("Graphs").font(.system(size: 14))
                            Image(systemName: "chart.bar")
                                .font(.system(size: 16))
                        }
                        .foregroundColor(SansColors.gray300)
                        .padding(.horizontal, 8).padding(.vertical, 6)
                    }
                    .buttonStyle(.plain)
                }
                .padding(.horizontal, 8)
                .padding(.vertical, 8)
            }

            Button(action: onToggleHeader) {
                Image(systemName: showHeader ? "chevron.up" : "chevron.down")
                    .font(.system(size: 12))
                    .foregroundColor(SansColors.gray500)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 2)
            }
            .buttonStyle(.plain)
        }
        .frame(maxWidth: .infinity)
        .background(SansColors.darkBackground.ignoresSafeArea(edges: .top))
        .animation(.easeInOut(duration: 0.3), value: showHeader)
    }
}

/// Fixed bottom navigation bar, mirroring `BottomNavBar` in `AppChrome.kt`.
/// Order: Home (1), Results (3), Graphs (2), Tables (0).
struct BottomNavBar: View {
    let currentPage: Int
    let onSelect: (Int) -> Void

    var body: some View {
        VStack(spacing: 0) {
            Rectangle()
                .fill(SansColors.navBorder)
                .frame(height: 1)
            HStack {
                navItem(1, "Home", "house")
                navItem(3, "Results", "list.bullet.rectangle")
                navItem(2, "Graphs", "chart.bar")
                navItem(0, "Tables", "doc.text")
            }
            .frame(maxWidth: .infinity)
            .padding(.vertical, 6)
        }
        .frame(maxWidth: .infinity)
        .background(SansColors.darkBackground.ignoresSafeArea(edges: .bottom))
    }

    private func navItem(_ page: Int, _ label: String, _ symbol: String) -> some View {
        let active = currentPage == page
        let color = active ? SansColors.primaryCyan : SansColors.gray500
        return Button {
            onSelect(page)
        } label: {
            VStack(spacing: 2) {
                Image(systemName: symbol)
                    .font(.system(size: 20))
                Text(label)
                    .font(.system(size: 10, weight: .medium))
            }
            .foregroundColor(color)
            .frame(maxWidth: .infinity)
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
        }
        .buttonStyle(.plain)
    }
}
